compiler/src/iree/compiler/Codegen/SPIRV/VideoCoreConfig.cpp

```diff
@@ -32,21 +42,151 @@ static LogicalResult setVideoCoreMatmulConfig(linalg::LinalgOp op,
+
+  int ocIndex;
+  if (!convDims.outputChannel.empty()) {
+    assert(convDims.outputChannel.size() == 1);
+    ocIndex = convDims.outputChannel.front();
+  } else if (!convDims.depth.empty()) {
+    // For depthwise convolution ops with multipler 1, we have the same
+    // input/filter/output channel size, which is being categorized as the
+    // multipler.
+    assert(convDims.depth.size() == 1);
+    ocIndex = convDims.depth.front();
+  } else {
+    // For pooling ops, the input/output channel size will be categorized
+    // as the additional batch dimension.
+    assert(convDims.batch.size() == 2);
+    ocIndex = convDims.batch.back();
+  }
+
+  Type outputType = op.getDpsInitOperand(0)->get().getType();
+  ArrayRef<int64_t> outputShape = llvm::cast<ShapedType>(outputType).getShape();
+  if ((convDims.inputChannel.empty() ||
+       ShapedType::isDynamic(convDims.inputChannel.front())) ||
+      llvm::any_of(outputShape.drop_front(), ShapedType::isDynamic)) {
+    return failure();
+  }
+
+  // Output tilings for the SPIRV Vectorize Pipeline
+  TileSizesListType tileSizes;
+  const int64_t outputChannelSize = loopRanges[ocIndex];
+
+  const bool isNCHW = ocIndex < ohIndex;
+  if (!isNCHW) {
+    // TODO(ROO-89): implement the NHWC case
+    return failure();
+  }
+
+  // Workgroup tiling is very simple for large OC tensors we just use the
+  // multiple of threads over channel
+  SmallVector<int64_t> workgroupTiling(4, 1); // (N, OC, OH, OW)
+  workgroupTiling[3] = getWorkgroupTiling(threadsPerWorkGroup, ow);
+  workgroupTiling[2] = getWorkgroupTiling(threadsPerWorkGroup, oh);
+  workgroupTiling[1] =
+      getWorkgroupTiling(threadsPerWorkGroup, outputChannelSize);
+  tileSizes.push_back(workgroupTiling);
+
+  // Remember OC->Z, OW->Y and OH->X
+  SmallVector<int64_t, 3> workgroupSize(3, 1); // (X, Y, Z)
+  workgroupSize[0] = std::max(workgroupTiling[3] / 4, 1ll);
+  workgroupSize[1] = workgroupTiling[2];
+  workgroupSize[2] = std::max(workgroupTiling[1] / 4, 1ll);
+
+  // Thread tiling seems to be most optimal when the last dim is tiled.
+  SmallVector<int64_t> threadTiling = {1, 1, 1,
+                                       optimalThreadTiling}; // (N, OC, OH, OW)
+  tileSizes.push_back(threadTiling);
+
+  // optimal tiling is on IC, F, F
+  SmallVector<int64_t> reductionTiling(loopRanges.size(), 0);
+  reductionTiling[convDims.inputChannel.front()] = 4;
+  reductionTiling[convDims.filterLoop.front()] = 1;
+  reductionTiling[convDims.filterLoop.back()] = 1;
+  tileSizes.push_back(reductionTiling);
+
+  // Tile along OH by size 1 to enable downsizing 2-D convolution to 1-D.
+  SmallVector<int64_t> windowTileSizes(4, 0);
+  windowTileSizes[ohIndex] = 1;
+  tileSizes.push_back(windowTileSizes);
+
+  auto pipeline = CodeGenPipeline::SPIRVBaseVectorize;
+  auto funcOp = op->getParentOfType<mlir::FunctionOpInterface>();
+  return setOpConfigAndEntryPointFnTranslation(funcOp, op, tileSizes, pipeline,
```

Comment:
Can you expand the comments here a bit more? Think of David for example who tries to understand this important part